runtime/autoload/provider/pythonx.vim

```diff
@@ -12,26 +12,31 @@ function! provider#pythonx#Detect(ver) abort
         \ 'g:python_host_skip_check' : 'g:python3_host_skip_check'
   let skip = exists(skip_var) ? {skip_var} : 0
   if exists(host_var)
-    " Disable auto detection
-    let [check, err] = s:check_interpreter({host_var}, a:ver, skip)
-    return check ? [{host_var}, err] : ['', err]
+    " Disable auto detection.
+    let [check, err, _] = s:check_interpreter({host_var}, a:ver, skip)
+    if check
+      return [{host_var}, err]
+    endif
+    return ['', 'provider#pythonx#Detect: could not load Python '.a:ver
+          \ .' (from '.host_var.'): '.err]
   endif
 
   let detect_versions = (a:ver == 2) ?
-        \   ['2.7', '2.6', '2', '']
-        \ : ['3.5', '3.4', '3.3', '3.2', '3', '']
+        \   ['2', '2.7', '2.6', '']
+        \ : ['3', '3.5', '3.4', '3.3', '']
 
   for prog in map(detect_versions, "'python' . v:val")
-    let [check, err] = s:check_interpreter(prog, a:ver, skip)
+    let [check, err, ver] = s:check_interpreter(prog, a:ver, skip)
```

Comment:
You are right, it's not used currently, but could be used then for the main fix from https://github.com/neovim/neovim/pull/2735.